serviceWrappers/linux/kLiveController

```diff
@@ -21,48 +21,71 @@
 ###
 # extend callable functions from "functions" script
 ###
+NAME=kLiveController
+LOG_DIR=/var/log
+PID_DIR=/var/run
+USE_NVM="1"
+KLIVE_CONTROLLER_PREFIX="/opt/kaltura/liveController/latest"
+STATUS_FUNCTION=status
+
+
 source /etc/rc.d/init.d/functions
 
-NAME="kLiveController"
-KLIVE_CONTROLLER_PREFIX="/opt/kaltura/liveController/latest"
-NODE_PATH="$KLIVE_CONTROLLER_PREFIX/node_modules"
-FOREVER="$NODE_PATH/forever/bin/forever"
-APPLICATION_PATH="$KLIVE_CONTROLLER_PREFIX/lib/App.js"
-LOG_DIR="/var/log/liveController"
-PID_DIR="/var/run"
+if [ -r /etc/default/${NAME} ]; then
+    echo ". /etc/default/${NAME}"
+    . /etc/default/${NAME}
+fi
+
+if [ -r /etc/rc.d/init.d/functions ]; then
+   echo ". source /etc/rc.d/init.d/functions"
+   . /etc/rc.d/init.d/functions
+elif [ -r /lib/lsb/init-functions ]; then
+   echo ". /lib/lsb/init-functions"
+   . /lib/lsb/init-functions
+   STATUS_FUNCTION=status_of_proc
+else
+    echo "Error, no function found for checking if ${NAME} is running. either not running in CentOS/Ubuntu or wrong path"
+fi
+
+NODE_PATH="${KLIVE_CONTROLLER_PREFIX}/node_modules"
+FOREVER="${NODE_PATH}/forever/bin/forever"
+APPLICATION_PATH="${KLIVE_CONTROLLER_PREFIX}/lib/App.js"
 MIN_UPTIME="5000"
 SPIN_SLEEP_TIME="2000"
-export PATH=$PATH:$NODE_PATH/forever/bin:/usr/local/bin
+export PATH=$PATH:${NODE_PATH}/forever/bin
 HOME=${HOME:-~}
 
 loadNvm() {
-    # This loads nvm
-    # must cd/pushd due to a bug in /root/.nvm/nvm.sh call to nvm_find_nvmrc()  
-    pushd ${KLIVE_CONTROLLER_PREFIX}
-    [ -z "${NVM_DIR}" ] && NVM_DIR="${HOME}/.nvm"
-    if [ -s "${NVM_DIR}/nvm.sh" ]; then
-        source "${NVM_DIR}/nvm.sh"
-        if [ -s ".nvmrc" ] ; then
-            nvm use || nvm install
+    if [ "${USE_NVM}" = "1" ]; then
+        # This loads nvm
+        # must cd/pushd due to a bug in /root/.nvm/nvm.sh call to nvm_find_nvmrc()
+        pushd ${KLIVE_CONTROLLER_PREFIX}
+        [ -z "${NVM_DIR}" ] && NVM_DIR="${HOME}/.nvm"
+        if [ -s "${NVM_DIR}/nvm.sh" ]; then
+            source "${NVM_DIR}/nvm.sh"
+            if [ -s ".nvmrc" ] ; then
+                nvm use || nvm install
+            else
+                echo "no .nvmrc in ${KLIVE_CONTROLLER_PREFIX}, Exiting"
+                exit 1
+            fi
         else
-            echo "no .nvmrc in ${KLIVE_CONTROLLER_PREFIX}, Exiting"
+            echo "nvm not found in ${NVM_DIR}, this is a must, Exiting!"
             exit 1
         fi
-    else
-        echo "nvm not found in ${NVM_DIR}, this is a must, Exiting!"
-        exit 1
+        popd
     fi
-    popd
 }
 
 start() {
-    [ -d ${LOG_DIR} ] || mkdir -p ${LOG_DIR}
+    echo "starting liveController..."
+    [ -d ${LOG_DIR}/liveController ] || mkdir -p ${LOG_DIR}/liveController
```

Comment:
liveController is concatenated to  LOG_DIR all around... so maybe it's better that the LOG_DIR will point to the this path